app/Http/Controllers/Admin/StandAdminController.php

```diff
@@ -84,30 +88,104 @@ public function createNewStand(Airfield $airfield, StandCreateRequest $request):
     {
         $validatorsInUse = $airfield->stands->pluck('identifier');
         if ($validatorsInUse->contains($request->get('identifier'))) {
-            return response()->json(['message' => 'Stand identifier in use for airfield.'], 409);
+            return response()->json(self::INVALID_IDENTIFIER_ERROR, 409);
         }
 
         // form request will validate existence of terminal if specified.
-        if ($terminal = Terminal::find($request->get('terminal_id'))) {
-            if ($terminal->airfield_id != $airfield->id) { // NOSONAR (cant merge the if statements, despite what sonar says!)
-                return response()->json(['message' => 'Invalid terminal for airfield.'], 400);
-            }
+        if (!$this->checkForTerminalValidity($request->get('terminal_id', null), $airfield->id)) {
+            return response()->json(self::INVALID_TERMINAL_ERROR, 400);
         }
 
+        $stand = Stand::create($this->formatObjectForStandFromRequest($request, $airfield->id));
+
+        return response()->json(['stand_id' => $stand->id], 201);
+    }
+
+    /**
+     * Modify a stand which is contained within a given airfield.
+     *
+     * @param Airfield $airfield
+     * @param Stand $stand
+     * @param StandCreateRequest $request
+     * @return JsonResponse
+     */
+    public function modifyStand(Airfield $airfield, Stand $stand, StandCreateRequest $request) : JsonResponse //NOSONAR
+    {
+        if ($stand->airfield_id != $airfield->id) {
+            return response()->json(self::STAND_NOT_IN_AIRFIELD_ERROR, 404);
+        }
+
+        $validatorsInUse = $airfield->stands->pluck('identifier');
```

Comment:
Would this prevent your from you to re-saving the stand with the same identifier, do we need to do a check to the effect of "not if the stand we're updating is the current one"?

Reply:
Good catch, I've amended with tests :)